### ISSUE TITLE
Inference issue with yolov8_quantize_model

### ISSUE THREAD

**Issue by chandra-ps612** [ultralytics/ultralytics#14912]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello @glenn-jocher ,
yolov8_custom_model(Uncompressed OpenVINO model)  produces similar inference results on the test set as the PyTorch model. Albeit, there are no inference results i.e. blank detection after post-training quantization using NNCF on the same dataset.

**Minimum Reproducible Code**
```
model.export(format='openvino', imgsz=[640,640], device='cpu', half=False, simplify=False, dynamic=False)  # creates 'yolov8n_openvino_model/'

# Creating and Instantiating DataLoader
import os
import torch
import warnings
import openvino as ov
from PIL import Image
from torch.utils.data import Dataset, DataLoader
from torchvision import transforms
warnings.simplefilter('ignore', Image.DecompressionBombWarning)

class CustomDataset(Dataset):
    def __init__(self, images_dir, labels_dir, transform=None):
        self.images_dir = images_dir
        self.labels_dir = labels_dir
        self.transform = transform
        self.image_files = [f for f in os.listdir(images_dir) if f.endswith('.jpg')]
        
    def __len__(self):
        return len(self.image_files)
    
    def __getitem__(self, idx):
        img_name = self.image_files[idx]
        image_path = os.path.join(self.images_dir, img_name)
        label_path = os.path.join(self.labels_dir, img_name.replace('.jpg', '.txt'))
        
        # Load image
        image = Image.open(image_path).convert('RGB')
        org_w, org_h = image.size[0], image.size[1]
        
        # Transform label
        transformed_label = []
        with open(label_path, 'r') as f:
            coords = [coord.strip().split() for coord in f.readlines()]
            # print(f'Label before transformation:{coords}')
            for coord in coords:
                cls_id, Xmin, Ymin, Xmax, Ymax = int(coord[0]), int(coord[1]), int(coord[2]), int(coord[3]), int(coord[4])
                ratio_w, rato_h = (640/org_w), (640/org_h)
                Xmin *= ratio_w
                Ymin *= rato_h
                Xmax *= ratio_w
                Ymax *= rato_h
                transformed_label.append([cls_id, Xmin, Ymax, Xmin, Ymax])

        transformed_label = torch.tensor(transformed_label, dtype=int)

        if self.transform:
            image = self.transform(image)
        
        return image, transformed_label


# Define the transform
transform = transforms.Compose([transforms.Resize((640, 640)), transforms.ToTensor()])


# Path to validation dataset
dataset_root_path = root_path_to_validation_dataset
images_dir = os.path.join(dataset_root_path, 'images')
labels_dir = os.path.join(dataset_root_path, 'labels_voc')

# Instantiate uncompressed model
core = ov.Core()
model_path = path_to_uncompressed_yolov8_openvino_model
model = core.read_model(model_path)

# Instantiate the custom dataset to be used as calibration data
val_dataset = CustomDataset(images_dir=images_dir, labels_dir=labels_dir, transform=transform)
dataset_loader = DataLoader(val_dataset, batch_size=1, shuffle=True)
```
**Note-** 
I used dataset_loader as an iterator to examine image and labels further.  

- image.shape: torch.tensor([1, 3, 640, 640])
- image.dtype: torch.float32
- image is a tensor that contains normalized values between 0 and 1
- label.dtype : torch.int64

```
# Optimize model using NNCF Post-training Quantization API
import nncf  # Neural Network Compression Framework
# Step 1: Initialize transformation function
def transform_fn(data_item):
    """
    Quantization transform function. Extracts and preprocess input data from dataloader item for quantization.
    Parameters:
       data_item: Tuple with data item produced by DataLoader during iteration
    Returns:
        input_tensor: Input data for quantization
    """
    img = data_item[0].numpy().astype(np.uint8)
    input_tensor = image_to_tensor(img)
   
    return input_tensor

# Step 2: Initialize NNCF Dataset
calibration_dataset = nncf.Dataset(dataset_loader, transform_func=transform_fn)

ignored_scope = nncf.IgnoredScope(
            types=["Multiply", "Subtract", "Sigmoid"],  # Ignore operations
            names=["/model.22/dfl/conv/Conv",  # In the post-processing subgraph
                   "/model.22/Add",
                   "/model.22/Add_1"])

# Detection model
quantized_model = nncf.quantize(model, calibration_dataset, preset=nncf.QuantizationPreset.MIXED, target_device=nncf.TargetDevice.CPU, ignored_scope=ignored_scope)
quantized_model_save_path = path_to_save_quantized_model
ov.serialize(quantized_model, quantized_model_save_path)
```
```
images_dir = path_to_test_images_dir
quantized_result = path_to_save_quantize_model_inference_results_on_test_dataset
if not os.path.exists(quantized_inference_dir): os.makedirs(quantized_inference_dir, exist_ok=True)
images = [os.path.join(images_dir, image) for image in os.listdir(images_dir)]

# Read quantize model
quantized_model = core.read_model(quantized_model_save_path)

# Load model on CPU
quantized_compiled_model = core.compile_model(quantized_model, device.value)

def main ():
    start = time.time()
    for image in tqdm(images):
        img_name = os.path.basename(image)
        img = np.array(cv2.imread(image))
        detections = detect(img, quantized_compiled_model)[0]
        print(detections)
        pred_im = draw_results(detections, img, label_map)
        save_path = os.path.join(quantized_inference_dir, img_name)
        cv2.imwrite(save_path, pred_im)
    end = time.time()
    print(f'Latency of Quantized OpenVINO IR format model: {((end - start) / len(images))*1000} ms')

if '__main__' == __name__:
    main()
```
Reference: https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/yolov8-optimization/yolov8-object-detection.ipynb
Even, I tried without ignored_scope argument, but still getting the same results. Please help me out in this context. 

### Additional

**Dependencies Installed**
```
nncf==2.11.0                  
openvino==2023.2.0
ultralytics==8.1.22  
```

**Comment by Burhan-Q** [comment]
@chandra-ps612 it's actually **a lot** easier to get a quantized OpenVino YOLO model, you only need to include `int8=True` when exporting with Ultralytics and provide a YOLO-format `data.yaml` for your calibration data (you can find details about the data YAMLs in our docs, here's the [detect format](https://docs.ultralytics.com/datasets/detect/)). 

```python
from ultralytics import YOLO

model = YOLO("yolov8n.pt")
model.export(
    format='openvino', 
    imgsz=[640,640], 
    device='cpu', 
    int8=True, 
    dynamic=False,
    data="my_data.yaml",
)  # creates 'yolov8n_openvino_int8_model/'
```
There is a [table here](https://docs.ultralytics.com/modes/export/#export-formats) showing all the supported arguments for OpenVino exports. If you'd like to review the code used for quantizing OpenVino models, you can see the `engine/exporter.py` file for deeper insight.

https://github.com/ultralytics/ultralytics/blob/9f22f45189bd32efc14ddcb7b7b42a548b7aef6f/ultralytics/engine/exporter.py#L476-L512
